plugins/woocommerce/includes/admin/meta-boxes/views/html-product-attribute-inner.php

```diff
@@ -37,10 +37,17 @@
 
 				if ( 'select' === $attribute_taxonomy->attribute_type ) {
 					$attribute_orderby = ! empty( $attribute_taxonomy->attribute_orderby ) ? $attribute_taxonomy->attribute_orderby : 'name';
+				  /**
+				   * Filter the length (number of terms) rendered in the list.
+	               *
+				   * @since 8.5.0
+				   * @param int $term_limit The maximum number of terms to display in the list.
+				   */
+					$term_limit = apply_filters( 'woocommerce_admin_terms_metabox_datalimit', 50 );
```

Comment:
Could you wrap the call to `apply_filters` in `absint`? Like so: 


```suggestion
					$term_limit = absint( apply_filters( 'woocommerce_admin_terms_metabox_datalimit', 50 ) );
```